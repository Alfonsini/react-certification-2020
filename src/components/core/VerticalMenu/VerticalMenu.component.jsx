import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHome, faStar } from '@fortawesome/free-solid-svg-icons';

import { StyledUl, StyledLi, StyledLink, StyledLinkTitle } from './VerticalMenu.styles';

function VerticalMenu(props) {
  return (
    <StyledUl>
      <StyledLi expanded={props.expanded} active>
        <StyledLink to="/" expanded={props.expanded} className="disabledCursor">
          <FontAwesomeIcon icon={faHome} />
          <StyledLinkTitle textTruncate>Home</StyledLinkTitle>
        </StyledLink>
      </StyledLi>
      <StyledLi expanded={props.expanded}>
        <StyledLink to="/" expanded={props.expanded} className="disabledCursor">
          <FontAwesomeIcon icon={faStar} />
          <StyledLinkTitle textTruncate>Favorities</StyledLinkTitle>
        </StyledLink>
      </StyledLi>
    </StyledUl>
  );
}

export default VerticalMenu;
