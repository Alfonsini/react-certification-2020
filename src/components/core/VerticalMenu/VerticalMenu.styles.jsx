import { Link } from 'react-router-dom';
import styled from 'styled-components';

const StyledUl = styled.ul`
  font-size: 1.1rem;
  font-family: 'Montserrat', Helvetica, Arial, serif;
  font-weight: 400;
  overflow-y: hidden;
  overflow-x: hidden;
  list-style: none;
  margin: 0;
  padding: 0;
  padding-bottom: 0.75rem;
`;

const StyledLi = styled.li`
  position: relative;
  white-space: nowrap;
  margin-bottom: 18px;

  ${(props) =>
    props.active &&
    `
    z-index: 1;
  `};

  ${(props) =>
    props.active &&
    `
    a {
      color: #dcdcdc;
      background: linear-gradient(118deg, #7367f0, #7367f0b3);
      box-shadow: 0 0 10px 1px rgb(115 103 240 / 70%);
      border-radius: 4px;
    }
  `};

  ${(props) =>
    props.active &&
    !props.expanded &&
    `
  a {
    background: ${props.theme.backgroundSidebarOption};
    box-shadow: none;
  }
`};

  :first-child {
    margin-top: 0.5rem;
  }

  :hover a > * {
    transition: transform 0.25s;
    transform: translateX(5px);
  }
`;

const StyledLink = styled(Link)`
  color: #6d6f97;
  margin: 0 15px;
  padding: 10px 15px 10px 15px;
  line-height: 1.45;
  text-overflow: ellipsis;
  overflow: hidden;
  align-items: center;
  ${(props) => !props.expanded && `width: 50px`};
  ${(props) => props.expanded && `width: calc(260px - 35px)`};
  display: flex;
  text-decoration: none;

  svg {
    margin-right: 1.1rem;
  }
`;

const StyledShadowBottom = styled.div`
  display: none;
  position: absolute;
  z-index: 2;
  height: 50px;
  width: 100%;
  pointer-events: none;
  filter: blur(5px);
`;

const StyledLinkTitle = styled.span`
  ${(props) =>
    props.textTruncate &&
    `
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `}
`;

export { StyledUl, StyledLi, StyledLink, StyledShadowBottom, StyledLinkTitle };
