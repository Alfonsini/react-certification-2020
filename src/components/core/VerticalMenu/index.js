export { default } from './VerticalMenu.component';
