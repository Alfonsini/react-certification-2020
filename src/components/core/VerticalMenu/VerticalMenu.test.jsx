import React from 'react';
import renderer from 'react-test-renderer';

import VerticalMenu from '.';

beforeEach(() => {});

afterEach(() => {});

describe('VerticalMenu', () => {
  test('sidebar not expanded', () => {
    const tree = renderer.create(<VerticalMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('sidebar expanded', () => {
    const tree = renderer.create(<VerticalMenu expanded />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
