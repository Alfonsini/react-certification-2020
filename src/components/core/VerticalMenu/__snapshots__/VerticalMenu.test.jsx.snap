// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalMenu sidebar expanded 1`] = `
<ul
  class="sc-bdfBwQ eGcbsb"
  role="list"
>
  <li
    class="sc-gsTCUz hpjNcD"
  >
    <a
      class="sc-dlfnbm fvtZPP"
      expanded="1"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-home fa-w-18 "
        data-icon="home"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-eCssSg jCnlqQ"
      >
        Home
      </span>
    </a>
  </li>
  <li
    class="sc-gsTCUz jDzifl"
  >
    <a
      class="sc-dlfnbm fvtZPP"
      expanded="1"
      href="/favorites"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-eCssSg jCnlqQ"
      >
        Favorites
      </span>
    </a>
  </li>
</ul>
`;

exports[`VerticalMenu sidebar not expanded 1`] = `
<ul
  class="sc-bdfBwQ eGcbsb"
  role="list"
>
  <li
    class="sc-gsTCUz iUsqMz"
  >
    <a
      class="sc-dlfnbm hPocZh"
      expanded="0"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-home fa-w-18 "
        data-icon="home"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-eCssSg jCnlqQ"
      >
        Home
      </span>
    </a>
  </li>
  <li
    class="sc-gsTCUz jDzifl"
  >
    <a
      class="sc-dlfnbm hPocZh"
      expanded="0"
      href="/favorites"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
        />
      </svg>
      <span
        class="sc-eCssSg jCnlqQ"
      >
        Favorites
      </span>
    </a>
  </li>
</ul>
`;
